### PR TITLE
Adjust mount-server auth flow to authenticate pipeline extension

### PR DESCRIPTION
**Pull request by kevyang**
Updates the auth flow in the Jupyter extension to accommodate authentication when running in sidecar mode. Previously, the pipeline extension only works unauthenticated or with the mount-server running in subprocess mode because the Jupyter extension itself did not have access to the session token that the mount-server negotiates upon login when the processes do not share the same container. This change updates the auth flow so that the mount-server auth is done in two steps, returning a redirect to the login page in the first and returning the Pachyderm configuration along with the session token in the second step. This enables the Jupyter extension to write the same configuration into its own container and use those for creating pipelines in Pachyderm deployments with authentication enabled.

Jira: [INT-895]

[INT-895]: https://pachyderm.atlassian.net/browse/INT-895?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (smalyala) [inline jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:190]: Does this need to be awaited?

**Review by smalyala — APPROVED** [review]
Left a question but otw LGTM!

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (kevyang) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:190]: no, we specifically do not want to await this since we want to redirect the user to the login page without waiting for their login to complete. i believe `auth_login_token` will only finish once the user completes the login. this is similar to what is done before within the mount server by creating the authenticate go routine and updating the token without waiting on it.

Comment (kevyang) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:190]: perhaps i should capture this in a comment

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (chainlink) [inline jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:17]: I believe the `$NB_USER` should have the current username https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#user-related-configurations

Comment (kevyang) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py]: ah good call. is this preferable to the suggestion that @chainlink made?

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (kevyang) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py]: i don't think so, but i could be wrong. any time we overwrite it, we would simply be updating it with the configuration that exists in the mount-server, and i can't think of a situation where we would want the configurations to be different (at least not yet).

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (kevyang) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:197]: good point. i am thinking we can just use response.rethrow() for this as that seems like what we would want to do after reading through the documentation. but i am not really an expert on python http so let me know if this is not correct.

**Review by brendoncarroll — APPROVED** [review]
I updated CODEOWNER so it won't auto-assign PFS in the future.
https://github.com/pachyderm/pachyderm/pull/8778

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (bbonenfant) [reply jupyter-extension/jupyterlab_pachyderm/mount_server_client.py]: Should we have a check at least that we are in sidecar mode? At the moment if I run this locally wont it overwrite my config.json?

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (bbonenfant) [inline jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:226]: I would use a `with` context here rather than manually closing the file

File: jupyter-extension/jupyterlab_pachyderm/mount_server_client.py
Comment (bbonenfant) [inline jupyter-extension/jupyterlab_pachyderm/mount_server_client.py:224]: it appears based on its usage that `pach_config_dir` is a file? Should it be named differently? Nonetheless it would be helpful to have a docstring that documents these arguments. The above comment can be moved into the docstrnig as well

**Review by bbonenfant — COMMENTED** [review]
It might be worth breaking this `write_token_to_config` method out of the class (it doesn't look like it uses the `self` argument) so that we can write some tests for it. Also we should probably get another review from @chainlink on this new change

**Review by chainlink — APPROVED** [review]
Testing and working in both sidecar and non sidecar modes!

**Pull request merged**
